File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/controller/brandController.js
// com.pinyougou.controller: 定义控制器;                         传入模块
app.controller('brandController',function($scope,$controller,brandService){

    // 伪继承: 传递$scope来实现:{$scope:$scope}让当前$scope等于baseController模块中的$scope,让他们通用.
    $controller('baseController',{$scope:$scope});

    // 查询品牌列表
    $scope.findAll=function(){
        brandService.findAll().success(
            function(response){
                $scope.list=response ;	// 给列表变量赋值
            }
        );
    }

    //分页方法
    $scope.findPage=function(page,rows) {
        brandService.findPage(page,rows).success(
            function(response){
                $scope.list = response.rows ; // 显示当前页数据 rows:行.
                $scope.paginationConf.totalItems = response.total; // 更新总记录数, 给分页控件(变量);
            }

        );
    }

    //增加品牌
    // add: 方法名; post: 请求方法; (请求路径,请求变量);
    $scope.save=function(){
        var object=null ; // 定义一个对象
        if($scope.entity.id!=null){
            object=brandService.update($scope.entity) ;
        }else{
            object=brandService.add($scope.entity) ;
        }

        object.success(
            function(response){
                if(response.success) {
                    $scope.reloadList(); //调用方法刷新列表
                }else{
                    alert(response.message) ; //打印message显示失败原因
                }
            }
        );
    }

    // 查询实体
    $scope.findOne=function(id){
        brandService.findOne(id).success(
            function(response){
                $scope.entity=response;
            }
        );
    }

    // 删除品牌信息
    $scope.dele=function(){
        brandService.dele($scope.selectIds).success(
            function(response){
                if(response.success){
                    $scope.reloadList(); //调用方法刷新列表
                }else{
                    alert(response.message) ; //打印message显示失败原因
                }
            }
        );
    }

    $scope.searchEntity={};     // 变量初始化;
    // 条件查询
    $scope.search=function(page, rows){
        brandService.search(page, rows, $scope.searchEntity).success(
            function(response){
                $scope.list = response.rows ; // 显示当前页数据 rows:行.
                $scope.paginationConf.totalItems = response.total; // 更新总记录数, 给分页控件(变量);
            }
        );
    }



});

File: pinyougou_parent/pinyougou_manager_web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController     // 相当于@controller和@ResposeBody的结合远程与本地都注入
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService ;

    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){
        return brandService.findAll() ;
    }

    @RequestMapping("/findPage")
    public PageResult findPage(int page, int rows){
        return brandService.findPage(page , rows) ;
    }

    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand brand){    // @RequestBody: 获取键值对形式的请求体
     try {
         brandService.add(brand);
         return new Result(true, "新增成功!");
     }catch(Exception e){
         e.printStackTrace();
         return new Result(false, "新增失败!");
     }
    }

    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
        return brandService.findOne(id) ;
    }

    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand brand){
        try{
            brandService.update(brand);
            return new Result(true,"修改成功!") ;
        }catch(Exception e){
            e.printStackTrace();
            return new Result(false,"修改失败!") ;
        }
    }

    @RequestMapping("/delete")
    public Result delete(Long[] ids){
        try{
            brandService.delete(ids);
            return new Result(true, "删除成功") ;
        }catch(Exception e){
            e.printStackTrace();
            return new Result(false,"删除失败") ;
        }
    }

     /*
    查询品牌内容
     */
    @RequestMapping("/search")       // brand: 品牌信息; 当前页面; 每页记录数;
    public PageResult search(@RequestBody TbBrand brand, int page, int rows){
        return brandService.findPage(brand, page, rows) ;
    }

    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList() {
        return brandService.selectOptionList();
    }
}
